core/ostctools.c

```diff
@@ -97,7 +97,6 @@ void ostctools_import(const char *file, struct dive_table *divetable, struct tri
 	// Read dive's raw data, header + profile
 	if (fseek(archive, 456, 0) == -1) {
 		report_error(failed_to_read_msg, file);
-		free(uc_tmp);
```

Comment:
1) Obviously correct, but that raises the question why `uc_tmp` is allocated on the heap in the first place
2) For better encapsulation, the `free(ostcdive)` calls should probably be `free_dive(ostcdive)`

Reply:
I think I implemented both of these suggestions in my force push.
I'd appreciate another look :-)